src/lib/swapper/api.ts

```diff
@@ -133,6 +133,8 @@ export type BuildTradeInput = GetTradeQuoteInput & {
   wallet: HDWallet
 }
 
+export type FailureAmount = { asset: Asset; buyAmountCryptoBaseUnit: string }
```

Comment:
I might be misunderstanding what this represents, but I _think_ it's the expected output of the underlying transactions that are executed to get to the user's requested trade pair.

If so, I think we should re-name this concept (including the related selector) to be something more like `IntermediaryTransactionOutput` or something, as this both seems more accurate, and generic, as we might want to use this data to show the user the steps that the trade will execute, and they aren't necessarily failures.

Reply:
yes happy to rename - i couldnt think of a better name so thank you for finding a better one.

In the case of future functionality for per-step executions, its highly unlikely this will be fit for purpose and 99% we'll totally rewrite this interface, but no worries until then

Reply:
actually rethinking this, you're right in that once we have per-step monitoring this can be used for the lifi warning.

I've renamed